AmethystAPI/src/amethyst/runtime/mod_info/ModInfo.cpp

```diff
@@ -9,36 +9,52 @@ std::optional<Amethyst::ModInfo> Amethyst::ModInfo::getModInfoFromText(std::stri
         data = json::parse(text);
     }
     catch (std::exception e) {
-        Log::Warning("Failed to parse mod.json");
-        return std::nullopt;
+        Log::Error("[AmethystRuntime] Failed to parse mod.json, for {}\n Error: {}", modName, e.what());
+        throw std::exception();
     }
 
     // Verify all fields are correct in config.json
     if (!data["meta"].is_object()) {
-        Log::Warning("Required field \"meta\" should be of type \"object\" in mod.json");
-        return std::nullopt;
+        Log::Error("[AmethystRuntime] Required field \"meta\" should be of type \"object\" in mod.json, for {}", modName);
```

Comment:
Use the Assert macro, added recently, going to be using it through the project. Not gonna add this message to every instance